src/h5web/metadata-viewer/EntityInfo.tsx

```diff
@@ -1,30 +1,53 @@
 import React, { ReactElement } from 'react';
-import type { HDF5Entity } from '../providers/models';
+import type { MyHDF5Entity } from '../providers/models';
 import styles from './MetadataViewer.module.css';
-import { isDataset, isDatatype, hasSimpleShape } from '../providers/utils';
+import {
+  hasMySimpleShape,
+  isMyDataset,
+  isMyDatatype,
+  isMyLink,
+} from '../providers/utils';
 import { renderShapeDims } from './utils';
 import RawInspector from './RawInspector';
+import LinkInfo from './LinkInfo';
 
 interface Props {
-  entity: HDF5Entity;
+  entity: MyHDF5Entity;
 }
 
 function EntityInfo(props: Props): ReactElement {
   const { entity } = props;
-  const { collection } = entity;
-
-  const entityKind =
-    collection.charAt(0).toUpperCase() + collection.slice(1, -1);
+  const { id, name, kind, parents } = entity;
 
   return (
     <table className={styles.table}>
       <thead>
         <tr>
-          <th colSpan={2}>{entityKind} info</th>
+          <th colSpan={2}>{kind.charAt(0).toUpperCase() + kind.slice(1)}</th>
         </tr>
       </thead>
       <tbody>
-        {(isDataset(entity) || isDatatype(entity)) && (
+        {id !== undefined && (
```

Comment:
`id` is optional in `MyHDF5Entity`, as some types of `HDF5Link` don't have entity IDs.